src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -325,13 +325,16 @@ internal void Update(double number)
 
                 case AggregationType.Histogram:
                     {
-                        int i;
-                        for (i = 0; i < this.histogramBuckets.ExplicitBounds.Length; i++)
+                        int i = 0;
+                        if (this.histogramBuckets.RunningBucketCounts.Length != 1)
                         {
-                            // Upper bound is inclusive
-                            if (number <= this.histogramBuckets.ExplicitBounds[i])
+                            for (; i < this.histogramBuckets.ExplicitBounds.Length; i++)
                             {
-                                break;
+                                // Upper bound is inclusive
+                                if (number <= this.histogramBuckets.ExplicitBounds[i++])
```

Comment:
Why increment `i` here again?

Reply:
Oops, this was unintentional. I was contemplating changing it to a while loop